### PR TITLE
Config Param 'overwritecondaddr' not working

### LINKED ISSUE #6914: Config Param 'overwritecondaddr' not working
### Steps to reproduce
1. clean install nextcloud
2. add conditional overwrite settings to config.php 
   'overwritehost' => 'my.domain.com',
   'overwritecondaddr' => '^192\.168\.0\.1$',

### Expected behaviour
Hostname should be overwritten when request comes in via 192.168.0.1, otherwise NOT.

### Actual behaviour
Hostname is always overwritten

### Server configuration

**Operating system**:
official Docker Image on different Docker Hosts

**Nextcloud version:** (see Nextcloud admin page)
12.0.3.3

**Updated from an older Nextcloud/ownCloud or fresh install:**
fresh

**Where did you install Nextcloud from:**
docker

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
<?php
$CONFIG = array (
  'htaccess.RewriteBase' => '/',
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'apps_paths' => 
  array (
    0 => 
    array (
      'path' => '/var/www/html/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 => 
    array (
      'path' => '/var/www/html/custom_apps',
      'url' => '/custom_apps',
      'writable' => true,
    ),
  ),
  'instanceid' => 'xxx',
  'passwordsalt' => 'xxx',
  'secret' => 'xxx',
  'trusted_domains' => 
  array (
    0 => '192.168.99.100:9000',
  ),
  'datadirectory' => '/var/www/html/data',
  'overwrite.cli.url' => 'http://192.168.99.100:9000',
  'dbtype' => 'sqlite3',
  'version' => '12.0.3.3',
  'installed' => true,
  
   'overwritecondaddr' => '^192\.168\.99\.11$',
   'overwritehost' => 'my.domain.com',
  
);
```
</details>

### Code causing Troubles...
Problem is in File .../lib/private/AppFramework/Http/Request.php in Method 'private function isOverwriteCondition($type = '')' ... type is empty when checking for e.g. Host-Override in Method 'getOverwriteHost' ... so with empty type check override method (isOverwriteCondition) always returns true cause of:

...
return $regex === '//' || preg_match($regex, $remoteAddr) === 1 || $type !== 'protocol';
...

-> type === '' ... so $type !== 'protocol' is always true ... so method always returns true in this case.

### PR DESCRIPTION
**Pull request by pavelkryl**
just ignoring/removing extra parameter 'protocol' as suggested by blizzz
Fix #6914

### CODE REVIEW

File: lib/private/AppFramework/Http/Request.php [hunk 1]
```diff
@@ -303,7 +303,7 @@ public function __unset($id) {
 	 * @return string
 	 */
 	public function getHeader(string $name): string {
-		$name = strtoupper(str_replace('-', '_', $name));
+		$name = strtoupper(str_replace('-', '_',$name));
```
Comment (blizzz) [inline lib/private/AppFramework/Http/Request.php:306]: please leave this space :) this should satisfy the codestyle linter test.

Comment (pavelkryl) [reply lib/private/AppFramework/Http/Request.php]: oh sorry, that wasn't on purpose

**Review by blizzz — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Comment by CarlSchwan** [comment]
Hi :) could you fix the DCO issue? You can find instructions here: https://github.com/nextcloud/server/pull/30654/checks?check_run_id=4808469925

The drone pipeline is failing but it's unrelated to this change

**Comment by szaimen** [comment]
@pavelkryl can you please fix DCO? Otherwise we cannot merge this change unfortunately :/

**Comment by blizzz** [comment]
continuation of this pr at #42930

**Pull request closed without merging**